### PR TITLE
2984 upload coordinates editor for annotations

### LINKED ISSUE #2984: Merge coordinate editor for annotations in c125_annotations
### Description

Updating the c125_annotations branch with the work made for the coordinate editor

This will include fix for the conflicts

### PR DESCRIPTION
**Pull request by MV88**
## Description
This pr updates c125_branch introducing geoemtry editor for annotations

## Issues
 - Fix #2984

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
No coordinate editor is implemented

**What is the new behavior?**
You can draw via map or form any type of annotation then if you select it you can edit it.

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: web/client/components/map/openlayers/Map.jsx [hunk 1]
```diff
@@ -159,6 +159,9 @@ class OpenlayersMap extends React.Component {
         if (this.props.registerHooks) {
             this.registerHooks();
         }
+        // setting default text for "Text" annotations
+        this.map.setProperties({defaultTextAnnotation: this.props.mapOptions && this.props.mapOptions.defaultTextAnnotation || "New"});
```
Comment (mbarto) [inline web/client/components/map/openlayers/Map.jsx]: the map component should not know anything about annotations

File: web/client/components/map/openlayers/DrawSupport.jsx [hunk 2]
```diff
@@ -8,14 +8,17 @@
...
+const {concat, head, find, slice, omit, isArray, last, filter} = require('lodash');
 const PropTypes = require('prop-types');
 const assign = require('object-assign');
 const uuid = require('uuid');
 const {isSimpleGeomType, getSimpleGeomType} = require('../../../utils/MapUtils');
 const {reprojectGeoJson, calculateDistance, reproject} = require('../../../utils/CoordinatesUtils');
-const VectorStyle = require('./VectorStyle');
 const wgs84Sphere = new ol.Sphere(6378137);
+const {transformPolygonToCircle} = require('../../../utils/DrawSupportUtils');
+const {isCompletePolygon} = require('../../../utils/AnnotationsUtils');
```
Comment (mbarto) [inline web/client/components/map/openlayers/DrawSupport.jsx]: DrawSupport should not know anything about annotations

File: web/client/localConfig.json [hunk 3]
```diff
@@ -216,7 +216,8 @@
                   },
                   "attribution": {
                     "container": "#mapstore-map-footer-container"
-                  }
+                  },
+                  "defaultTextAnnotation": "custom text"
```
Comment (mbarto) [inline web/client/localConfig.json]: It's not a map property / option, it is annotations related

File: web/client/components/map/openlayers/Map.jsx (hunk 1 above)
Comment (mbarto) [inline web/client/components/map/openlayers/Map.jsx]: This is annotations related, the map should not be aware of it

Comment (MV88) [reply web/client/components/map/openlayers/Map.jsx]: this is a duplicated comment

**Pull request merged**